hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileInfo.java

```diff
@@ -516,7 +519,7 @@ public static boolean isMobRefFile(final Path path) {
     parts[0] = fileName.substring(0, lastIndex);
     parts[1] = fileName.substring(lastIndex + 1);
     String name = parts[0] + "." + parts[1];
-    Matcher m = REF_NAME_PATTERN.matcher(name);
+    Matcher m = MOB_REF_NAME_PATTERN.matcher(name);
     return m.matches() && m.groupCount() > 1;
```

Comment:
The method isMobRefFile has issue:
- if mob file was reference file, would return true
eg:`t02_mob=34916d9a943a3b1fdb4a2bbeef2e3e97-d41d8cd98f00b204e9800998ecf8427e20241112464aebcdfc354fbb834d58900de1226d_6424974cfc2ba56e1c6f43c34e833a8e` 
- if mob file was not reference file, would still true
eg:`d41d8cd98f00b204e9800998ecf8427e202411128ebcda68f97943cb80b64a00a0841a8d_6424974cfc2ba56e1c6f43c34e833a8e`

So, I update it, when the mob file is reference file, it would return true, otherwise return false.
And add a new UT for this method (`TestMobFileType`).